### PR TITLE
feat: add more details about the sst in sst-metadata tool

### PR DESCRIPTION
**Pull request by zouxiang1993**
## Rationale
More details about the sst are neeeded for troubleshooting problems.

## Detailed Changes
- Output some statistics about the file;
- Output compression information;

## Test Plan
Check the output of sst-meta tool.

### CODE REVIEW

**Comment by zouxiang1993** [comment]
Output example: 

FileStatistics {
	file_count: 36,
	size: 214.65,
	metadata_size: 25.34, 
	kv_size: 24.14,
	filter_size: 17.71,
	row_num: 2175886,
}
FieldStatistics: 
serviceVersion,	 compressed_size: 3.32mb,	 uncompressed_size: 7.75mb,	 compress_ratio: 2.33
extra6,	 compressed_size: 0.02mb,	 uncompressed_size: 0.02mb,	 compress_ratio: 0.75
subtag,	 compressed_size: 8.89mb,	 uncompressed_size: 21.37mb,	 compress_ratio: 2.41
tsid,	 compressed_size: 19.65mb,	 uncompressed_size: 19.98mb,	 compress_ratio: 1.02
extra4,	 compressed_size: 0.17mb,	 uncompressed_size: 0.37mb,	 compress_ratio: 2.19

**Comment by ShiKaiWi** [comment]
@zouxiang1993 I'm OK with this PR but @jiacai2050 is the owner of this module, I guess we should wait for his review.

**Comment by jiacai2050** [comment]
It seems those info are duplicated with before?

```
FileStatistics {
	file_count: 36,
	size: 214.65,
	metadata_size: 25.34, 
	kv_size: 24.14,
	filter_size: 17.71,
	row_num: 2175886,
}
```

**Comment by zouxiang1993** [comment]
> It seems those info are duplicated with before?
> 
> ```
> FileStatistics {
> 	file_count: 36,
> 	size: 214.65,
> 	metadata_size: 25.34, 
> 	kv_size: 24.14,
> 	filter_size: 17.71,
> 	row_num: 2175886,
> }
> ```
> 
> Original output
> 
> ```
> Location:2097.sst, time_range:[2023-05-09 06:00:00, 2023-05-09 08:00:00), max_seq:132734153, size:348.675M, metadata:40.142M, kv:38.175M, filter:28.525M, row_num:7038949
> ```

Yes. But these info are statistics among all the sst file.

File: tools/src/bin/sst-metadata.rs [hunk 1]
```diff
@@ -91,6 +123,52 @@ async fn run(args: Args) -> Result<()> {
...
+        }
+        println!("{}", file_stats.to_string());
+
+        println!("FieldStatistics: ");
+        for (k, v) in field_stats_map.iter() {
+            println!("{},\t compressed_size: {:.2}mb,\t uncompressed_size: {:.2}mb,\t compress_ratio: {:.2}",
+                     k,
+                     as_mb(v.compressed_size as usize),
+                     as_mb(v.uncompressed_size as usize),
+                     v.uncompressed_size as f64 / v.compressed_size as f64);
```
Comment (jiacai2050) [inline tools/src/bin/sst-metadata.rs:167]: How to explain this?

From docs, compression_size represent pages with compression, and uncompressed_size represent page without uncompression, it seems they have no relation at all.

File: tools/src/bin/sst-metadata.rs [hunk 2]
```diff
@@ -91,6 +123,52 @@ async fn run(args: Args) -> Result<()> {
         metas.push(meta);
     }
 
+    if args.stats {
+        let mut file_stats = FileStatistics::default();
+        let mut field_stats_map = HashMap::new();
+        for (object_meta, sst_metadata, metadata_size, kv_size) in metas {
```
Comment (jiacai2050) [inline tools/src/bin/sst-metadata.rs:129]: It seems those parse is duplicated with before, a better way is to collect those size info here
https://github.com/CeresDB/ceresdb/blob/9a9c0f79116dff6df2a09fc8766e6952fd5f4432/tools/src/bin/sst-metadata.rs#L122

and print those size at last.

File: tools/src/bin/sst-metadata.rs (hunk 1 above)
Comment (zouxiang1993) [reply tools/src/bin/sst-metadata.rs]: uncompressed_size is the size before compress, compressed_size is the size after compress.

File: tools/src/bin/sst-metadata.rs [hunk 3]
```diff
@@ -114,6 +144,28 @@ async fn run(args: Args) -> Result<()> {
...
+        file_stats.kv_size += kv_size;
+        file_stats.filter_size += filter_size;
+        file_stats.row_num += row_num;
+
+        let fields = file_metadata.schema().get_fields();
+        for (_, row_group) in parquet_meta.row_groups().iter().enumerate() {
+            for i in 0..fields.len() {
+                let column_meta = row_group.column(i);
+                let field_name = fields.get(i).unwrap().get_basic_info().name().to_string();
+                if !field_stats_map.contains_key(&field_name) {
```
Comment (jiacai2050) [inline tools/src/bin/sst-metadata.rs:160]:
```
let mut field_stats = field_stats_map.entry(field_name).or_insert_with(FieldStatistics::default());
```

File: tools/src/bin/sst-metadata.rs (hunk 2 above)
Comment (zouxiang1993) [reply tools/src/bin/sst-metadata.rs]: fixed

**Review by jiacai2050 — APPROVED** [review]

**Pull request merged**